ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/HistoricalBlockHashProcessor.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright contributors to Hyperledger Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import static org.hyperledger.besu.evm.operation.BlockHashOperation.HISTORY_SERVE_WINDOW;
+
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.ethereum.chain.Blockchain;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.evm.account.MutableAccount;
+import org.hyperledger.besu.evm.operation.BlockHashOperation;
+import org.hyperledger.besu.evm.worldstate.WorldUpdater;
+import org.hyperledger.besu.plugin.data.ProcessableBlockHeader;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.tuweni.units.bigints.UInt256;
+
+/**
+ * Processes and stores historical block hashes in accordance with EIP-2935. This class is
+ * responsible for managing the storage of block hashes to support EIP-2935, which introduces
+ * historical block hash access in smart contracts.
+ */
+public class HistoricalBlockHashProcessor {
+
+  public static final Address HISTORY_STORAGE_ADDRESS = BlockHashOperation.HISTORY_STORAGE_ADDRESS;
```

Comment:
nit: Why not just use the constant from `BlockHashOperation` directly? It's already public

Reply:
Because it is not available from some places it is needed, like MainnetProtocolSpecs, without introducing a circular dependency.